FinnUI/Sources/Fullscreen/NotificationCenter/Views/NotificationCenterView.swift

```diff
@@ -0,0 +1,304 @@
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public override func layoutSubviews() {
+        if segmentContainers == nil {
+            setupSegmentedControl()
+        }
+        
+        super.layoutSubviews()
+    }
+}
+
+// MARK: - Public Methods
+public extension NotificationCenterView {
+    func reloadData() {
+        segmentedControl.removeAllSegments()
+        segmentContainers?.forEach { $0.tableView.removeFromSuperview() }
+        setupSegmentedControl()
+    }
+    
+    func reloadRows(at indexPaths: [IndexPath], inSegment segment: Int) {
+        guard let segmentContainers = segmentContainers, segmentContainers.indices ~= segment else {
+            return
+        }
+        
+        segmentContainers[segment].tableView.reloadRows(at: indexPaths, with: .automatic)
+    }
+    
+    func indexPathForSelectedRow(inSegment segment: Int) -> IndexPath? {
+        guard let segmentContainers = segmentContainers, segmentContainers.indices ~= segment else {
+            return nil
+        }
+        
+        return segmentContainers[segment].tableView.indexPathForSelectedRow
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension NotificationCenterView: UITableViewDataSource {
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        dataSource?.notificationCenterView(self, numberOfSectionsInSegment: selectedSegment) ?? 0
+    }
+    
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        dataSource?.notificationCenterView(self, segment: selectedSegment, numberOfRowsInSection: section) ?? 0
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cellType = dataSource?.notificationCenterView(self, segment: selectedSegment, modelForCellAt: indexPath)
+        let timestamp = dataSource?.notificationCenterView(self, segment: selectedSegment, timestampForCellAt: indexPath)
+        let overflow = dataSource?.notificationCenterView(self, segment: selectedSegment, overflowInSection: indexPath.section) ?? false
+        let isLast = indexPath.row == tableView.numberOfRows(inSection: indexPath.section) - 1
+        
+        switch cellType {
+        case let .notificationCell(model):
+            let cell = tableView.dequeue(NotificationCell.self, for: indexPath)
+            cell.remoteImageViewDataSource = remoteImageViewDataSource
+            cell.configure(with: model, timestamp: timestamp, hideSeparator: isLast, showGradient: isLast && overflow)
+            return cell
+        case let .feedbackCell(delegate, state, model):
+            let cell = tableView.dequeue(FeedbackCell.self, for: indexPath)
+            cell.delegate = delegate
+            cell.configure(for: state, with: model)
+            return cell
+        case .none:
+            fatalError("Cell type not specified")
```

Comment:
It would have been nice with an error log for this case. But you don't have to do it in this task since we do not have that infrastructure in place. But perhaps we should add a task to add it

Reply:
Sure, that would be good to have 👍